src/keyboard/mod.rs

```diff
@@ -0,0 +1,258 @@
+//! Utilities for keyboard handling
+//!
+//! This module provides utilities for keyboardand keymap handling: keymap interpretation
+//! and forwarding keystrokes to clients using xkbcommon.
+//!
+//! You can first create a `KbdHandle` using the `create_keyboard_handler` function in this module.
+//! The handle you obtained can be cloned to access this keyboard state from different places. It is
+//! expected that such a context is created for each keyboard the compositor has access to.
```

Comment:
Oh, ok, I understand now.

That's right. on this point we can hardly take inspiration from weston. They're going full C-style here, with pointers all over the place.